cli/cli/cmds_volume.go

```diff
@@ -339,7 +344,11 @@ func (c *CLI) initVolumeCmds() {
 			// volumes that are attached or can be attached should be valid
 			// candidates for a mount operation
 			result, err := c.lsVolumes(
-				args, volumeStatusAttached, volumeStatusAvailable)
+				args,
+				requestAttachments(
+					apitypes.VolumeAttachmentsMine,
+					apitypes.VolumeAttachmentsAttached,
+					apitypes.VolumeAttachmentsUnattached))
```

Comment:
Hi @codenrhoden,

I just noticed this after re-reading your comments:

> This makes sense on its own, but gives Mine precedence over Attached and Unattached.

Well, `mine` has no precedence over `unattached` as `mine` only has any effect if a volume has attachments.

And yes, `mine` has precedence over `attached`, but if you take a close look at the tables above you'll notice there is no attempt of the following:

> Request attachment information for all retrieved volumes and return only volumes that attached to the instance provided in the instance ID or are not attached to any instance at all or **are attached at all**.

The above would be what you described -- a mask that's the same as saying get all attachments for all volumes, right?

As it is, the logic is such to enable requesting all volumes attached to the provided instance or all volumes not attached to any instance, aka. volumes available to be attached.

The order of the server logic is specifically designed to enable this. That's why the `attached` bit only removes unattached volumes if the `unattached` bit is not set. That's why the `unattached` bit only removed attached volumes if the`attached` bit is not set.

Again, I do have [these tests](https://github.com/akutz/libstorage/blob/bugfix/attachment-filtering-logic/drivers/storage/vfs/tests/vfs_test.go#L166-L312) that validate the desired logic. 

I'm fully willing to admit I'm off base on all of this, but I also think it's possible I just did a **supremely** shitty job describing the purpose and implementation of these things with which to begin, and it's lead to much confusion.